src/iter/mod.rs

```diff
@@ -650,6 +656,19 @@ pub trait ParallelIterator: Sized {
         unzip::partition(self, predicate)
     }
 
+    /// Partitions and maps the items of a parallel iterator into a pair of
+    /// arbitrary `ParallelExtend` containers.  `Either::Left` items go into
+    /// the first container, and `Either::Right` items go into the second.
+    fn partition_map<A, B, P, L, R>(self, predicate: P) -> (A, B)
```

Comment:
This is meant for parity with itertools, I guess? Makes sense. I was wondering if we should link directly to itertools, so we can use the same `Either` type, but it's problematic that it is at only 0.6 I suppose, and it seems a *bit* silly to add an itertools dependency just for `Either`.

(cc @bluss -- as an aside, any reason itertools can't be 1.0?)